server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java

```diff
@@ -780,6 +791,42 @@ private Entry(
             assert assertShardsConsistent(this.source, this.state, this.indices, this.shards, this.shardStatusByRepoShardId);
         }
 
+        private Entry(
```

Comment:
This could use another follow-up to make this the only primary constructor since we now have two. I wanted this for efficiencies sake but didn't want to do that refactoring here since it requires turning all other constructors into static methods and adds a bunch of LoC changed (since there's so much logic in the other primary constructor).

Reply:
Maybe do that first?

Reply:
Sure how about https://github.com/elastic/elasticsearch/pull/89813 :)